### PR TITLE
unit tests for git-repository-provider

### PR DESCRIPTION
**Pull request by elaihau**
Signed-off-by: elaihau <liang.huang@ericsson.com>

### CODE REVIEW

**Comment by elaihau** [comment]
i didn't cover the functions that the multi root feature never touched.

**Comment by vince-fugnitto** [comment]
reviewing now :)

File: packages/git/src/browser/git-repository-provider.ts [hunk 1]
```diff
@@ -113,7 +113,8 @@ export class GitRepositoryProvider {
 
     async refresh(options?: GitRefreshOptions): Promise<void> {
         const roots: FileStat[] = [];
-        for (const root of await this.workspaceService.roots) {
```
Comment (akosyakov) [inline packages/git/src/browser/git-repository-provider.ts]: What is the difference? more code?

Comment (elaihau) [reply packages/git/src/browser/git-repository-provider.ts]: no difference from the functionality point of view, but it makes the code easier to unit test, as `tryGetRoots()` can be stubbed in the following way:
```ts
tryGetRoots.onCall(0).returns(one value);
tryGetRoots.onCall(1).returns(another value);
tryGetRoots.onCall(2).returns(and another value);
```
while `await ws.roots` is a getter from WorkspaceService, and 
```ts
roots.onCall(0).value(Promise.resolve(one value))
roots.onCall(1).value(Promise.resolve(another value))
roots.onCall(2).value(Promise.resolve(and another value))
```
did not works - only "and another value" from the 3rd line gets resolved.

Comment (akosyakov) [reply packages/git/src/browser/git-repository-provider.ts]: ok :)

**Pull request merged**